lib/cluster.js

```diff
@@ -155,42 +206,53 @@ function addListeners() {
 	listenersAdded = true;
 
 	if (cluster().isMaster) {
-		// Listen for worker responses to requests for local metrics
-		cluster().on('message', (worker, message) => {
-			if (message.type === GET_METRICS_RES) {
-				const request = requests.get(message.requestId);
-
-				if (message.error) {
-					request.done(new Error(message.error));
-					return;
-				}
+		// Listen for cluster responses to requests for local metrics
+		cluster().on('message', handleWorkerResponse);
+	}
+}
 
-				message.metrics.forEach(registry => request.responses.push(registry));
-				request.pending--;
+function getWorkerThreadsMetrics(message) {
+	workersQueue.forEach(worker => {
+		if (worker && worker instanceof Worker) {
+			const metricsChannel = new MessageChannel();
 
-				if (request.pending === 0) {
-					// finalize
-					requests.delete(message.requestId);
-					clearTimeout(request.errorTimeout);
+			worker.postMessage(
+				{
+					...message,
+					port: metricsChannel.port1,
+				},
+				[metricsChannel.port1],
+			);
 
-					const registry = AggregatorRegistry.aggregate(request.responses);
-					const promString = registry.metrics();
-					request.done(null, promString);
+			metricsChannel.port2.on('message', response => {
+				if (response.type === GET_METRICS_RES) {
+					if (cluster().isWorker) {
```

Comment:
I have no experience with worker threads. Can they be used (and are they ever practically used) in conjunction with the cluster module?